libvast/vast/system/source.hpp

```diff
@@ -109,29 +115,73 @@ struct source_state {
   /// The maximum number of events to ingest.
   caf::optional<size_t> remaining;
 
+  /// The import-local schema.
+  vast::schema local_schema;
+
+  /// The import type restriction.
+  vast::data type_restriction;
+
   /// Stores whether `reader` is constructed.
-  bool initialized;
+  bool reader_initialized;
+
+  /// Current metrics for the accountant.
+  measurement metrics;
 
   // -- utility functions ------------------------------------------------------
 
   /// Initializes the state.
-  void init(Reader rd, caf::optional<size_t> max_events) {
-    // Initialize members from given arguments.
+  void init(Reader rd, caf::optional<size_t> max_events,
+            type_registry_type type_registry, vast::schema sch,
+            vast::data type_restr, accountant_type acc) {
+    // Create the reader.
     name = reader.name();
     new (&reader) Reader(std::move(rd));
+    reader_initialized = true;
     remaining = std::move(max_events);
-    initialized = true;
+    local_schema = std::move(sch);
+    type_restriction = std::move(type_restr);
```

Comment:
```suggestion
    type_filter = std::move(type_restr);
```

This may need adjusting other stuff as well, but feels much easier to understand.